lib/src/components/buttons/button.dart

```diff
@@ -136,12 +162,32 @@ class ZetaButton extends StatelessWidget {
         onPressed: onPressed,
         style: buttonStyle(colors, borderType, type, null),
         child: SelectionContainer.disabled(
-          child: label.isEmpty
-              ? const SizedBox()
-              : Text(
+          child: Row(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              if (trailingIcon != null)
+                Icon(
+                  trailingIcon,
+                  size: _iconSize,
+                ),
+              if (label.isNotEmpty)
+                Text(
                   label,
                   style: _textStyle,
-                ).paddingHorizontal(_textPadding),
+                ),
+              if (leadingIcon != null)
+                Icon(
+                  leadingIcon,
+                  size: _iconSize,
+                ),
+            ]
+                .divide(
+                  const SizedBox(
```

Comment:
Compared to the design, these buttons are too wide; not sure why though?

Reply:
I'm not seeing the sizes being off , im judging this from the widgetbook but they look fine to me

Reply:
Button is rendered as 154.2px
![image](https://github.com/zebrafed/zeta_flutter/assets/23633655/7869c5a5-60b0-4e4f-bd6a-c8836e23a71a)
But should be 138px
![image](https://github.com/zebrafed/zeta_flutter/assets/23633655/23fc6855-4b6e-452d-a4e1-6228724c2b29)
This can be fixed by changing _textPadding (line 211):
```dart
  double get _textPadding {
    switch (size) {
      case ZetaWidgetSize.large:
        return ZetaSpacing.x4;

      case ZetaWidgetSize.medium:
        return ZetaSpacing.x3;

      case ZetaWidgetSize.small:
        return ZetaSpacing.x1;
    }
  }
```

Reply:
I've added those changes in